### PR TITLE
Fix regressions with password reset

### LINKED ISSUE #5564: Unable to reset password
This is a recent regression. When submitting the password reset form you are shown a "Please sign in" notification instead of the password being reset.

http://recordit.co/B3KOe9KW4W

There are also no functional tests in place for the reset form, it would be good to get those added so that we can catch regressions such as this.

### PR DESCRIPTION
**Pull request by ErisDS**
This fixes the issue, in case this is blocking people. I've marked it as a WIP because I still need to update the tests to ensure this can't regress again.

fixes #5564
- adds missing part of `/setup/` url in authentication middleware
- ensures data is passed through from API to model in correct (new) format for password reset

### CODE REVIEW

**Comment by ErisDS** [comment]
This has been updated with tests now and is good to go

File: core/test/integration/api/api_users_spec.js [hunk 1]
```diff
@@ -396,6 +396,7 @@ describe('Users API', function () {
             });
 
             sandbox.stub(mail, 'send', function () {
+                console.log('sending mail!!!');
```
Comment (halfdan) [inline core/test/integration/api/api_users_spec.js]: :ghost:

File: core/test/integration/api/api_authentication_spec.js [hunk 2]
```diff
@@ -204,51 +304,4 @@ describe('Authentication API', function () {
...
-    //                should.exist(err);
-
-    //                err.name.should.equal('NoPermissionError');
-    //                err.code.should.equal(403);
-
-    //                done();
-    //            });
-    //        });
-    //    });
-    // });
```
Comment (halfdan) [inline core/test/integration/api/api_authentication_spec.js]: Any reason for all the commented out tests? Can we make them work?

Comment (ErisDS) [inline core/test/integration/api/api_authentication_spec.js]: That block has been accidentally commented out for months! This PR re-adds them further up but I forgot to clean up the comment block apparently!

Comment (halfdan) [inline core/test/integration/api/api_authentication_spec.js]: ..and I fail at reading diffs. Need more :coffee:, shouldn't read diffs before my first.

Comment (ErisDS) [inline core/test/integration/api/api_authentication_spec.js]: Man I was soooo confused staring at the file and not seeing this!!!

**Comment by ErisDS** [comment]
Cleaned up the console log

**Comment by jaswilli** [comment]
https://travis-ci.org/TryGhost/Ghost/builds/71841940

**Pull request merged**